elasticgraph-warehouse/spec/integration/elastic_graph/warehouse/schema_definition/rake_tasks_spec.rb

```diff
@@ -0,0 +1,219 @@
+# Copyright 2024 - 2025 Block, Inc.
+#
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+#
+# frozen_string_literal: true
+
+require "elastic_graph/constants"
+require "elastic_graph/schema_definition/rake_tasks"
+require "elastic_graph/warehouse/schema_definition/api_extension"
+require "yaml"
+
+module ElasticGraph
+  module Warehouse
+    module SchemaDefinition
+      RSpec.describe "Warehouse RakeTasks", :rake_task, :in_temp_dir do
+        describe "schema_artifacts:dump" do
+          it "dumps warehouse artifact when warehouse tables are defined" do
+            write_warehouse_schema(table_defs: <<~EOS)
+              s.object_type "Product" do |t|
+                t.field "id", "ID"
+                t.field "name", "String"
+                t.field "price", "Float"
+                t.index "products" do |i|
+                  i.warehouse_table "products"
+                end
+              end
+            EOS
+
+            expect {
+              output = run_rake_with_warehouse("schema_artifacts:dump")
+              expect(output.lines).to include(
+                a_string_including("Dumped", Warehouse::DATA_WAREHOUSE_FILE)
+              )
+            }.to change { read_artifact(Warehouse::DATA_WAREHOUSE_FILE) }
+              .from(a_falsy_value)
+              .to(a_string_including("products:", "table_schema:", "CREATE TABLE IF NOT EXISTS products"))
+
+            warehouse_config = YAML.safe_load(read_artifact(Warehouse::DATA_WAREHOUSE_FILE))
+            expect(warehouse_config.keys).to eq(["tables"])
+            expect(warehouse_config["tables"].keys).to eq(["products"])
+            expect(warehouse_config["tables"]["products"]).to have_key("table_schema")
+            expect(warehouse_config["tables"]["products"]["table_schema"]).to start_with("CREATE TABLE IF NOT EXISTS products")
+          end
+
+          it "does not dump warehouse artifact when no warehouse tables are defined" do
+            write_warehouse_schema(table_defs: <<~EOS)
+              s.object_type "User" do |t|
+                t.field "id", "ID"
+                t.field "name", "String"
+                # No warehouse_table definition
+              end
+            EOS
+
+            output = run_rake_with_warehouse("schema_artifacts:dump")
+
+            expect(output.lines).not_to include(
+              a_string_including("Dumped", Warehouse::DATA_WAREHOUSE_FILE)
+            )
```

Comment:
```suggestion
            expect(output).to exclude(DATA_WAREHOUSE_FILE)
```

- We have `exclude` [defined as the negation](https://github.com/block/elasticgraph/blob/2dc3a3aae2ac2814ec688923ffcf93de10b9d832/spec_support/spec_helper.rb#L243) of `include`; might as well use it.
- With a negative expectation like this it's useful to be "stricter" and to exclude any mention of the warehouse file. (What you had would have allowed the warehouse file to be mentioned so long as the word "Dumped" wasn't on the same line.)
- You can just use `DATA_WAREHOUSE_FILE` throughout this file rather than `Warehouse::DATA_WAREHOUSE_FILE` because you are within the `ElasticGraph::Warehouse` module. (The entire reason to put `RSpec.describe` within the module declarations is to allow yourself to access the constants w/o needing to fully qualify them...).